File: 3354_Make_Array_Elements_Equal_to_Zero/solution.c
int countValidSelections(int* nums, int numsSize) {
    int total = 0;
    for (int i = 0; i < numsSize; i++) {
        total += nums[i];
    }

    int left_sum = 0, count = 0;
    for (int i = 0; i < numsSize; i++) {
        if (nums[i] == 0) {
            int right_sum = total - left_sum;
            int diff = abs(left_sum - right_sum);
            count += (diff == 0) ? 2 : (diff == 1) ? 1 : 0;
        }
        left_sum += nums[i];
    }

    return count;
}

File: 3354_Make_Array_Elements_Equal_to_Zero/solution.py
class Solution:
    def countValidSelections(self, nums: List[int]) -> int:
        total = sum(nums)
        left_sum, count = 0, 0
        
        for num in nums:
            if num == 0:
                right_sum = total - left_sum
                diff = abs(left_sum - right_sum)
                count += 2 if diff == 0 else (1 if diff == 1 else 0)
            left_sum += num
        
        return count